auth-tokens/src/main/java/com/palantir/tokens/auth/UnverifiedJsonWebToken.java

```diff
@@ -85,6 +88,14 @@ public abstract class UnverifiedJsonWebToken {
     @Value.Parameter
     public abstract Optional<String> getUnverifiedOrganizationId();
 
+    /**
+     * Returns the unverified expiration time, i.e. the "exp" claim, of the JWT
+     * or absent if the JWT does not contain the "exp" claim.
+     */
+    @Safe
+    @Value.Parameter
+    public abstract Optional<OffsetDateTime> getUnverifiedExpirationTime();
```

Comment:
I'm very wary about exposing this, for a couple reasons.
- The `exp` claim suggests to consumers that the token will be valid until that time. But this is not the case because tokens can be explicitly revoked before then. This is very common with short-lived API tokens generated for compute jobs, etc. This API is extremely easy to misuse in a way that will result in consumed accepting revoked tokens as valid.
- Not all tokens have `exp` claims, so consumers are still going to have to deal with the case when this claim is absent.